src/Type/TypeCombinator.php

```diff
@@ -703,6 +707,10 @@ private static function processArrayAccessoryTypes(array $arrayTypes): array
 			$commonAccessoryTypes[] = $accessoryType[0];
 		}
 
+		if (TrinaryLogic::createYes()->and(...$isIterableAtLeastOnce)->yes()) {
+			$commonAccessoryTypes[] = new NonEmptyArrayType();
```

Comment:
atm this can lead to some arrays, which while processing have the `NonEmptyArrayType` 2-times.
when `TypeCombinator::union()` finally ends, this gets normalized again.

we could check at this if here, whether `commonAccessoryTypes` already contains `NonEmptyArrayType` so we don't add it a 2nd time. 
for readability I have chosen not todo this additionaly check. but I am fine with adding it, in case you think we should do it.